Parser/Types/TypeAnalyze.cs

```diff
@@ -72,9 +84,10 @@ public override NType Visit (NWriteStmt w)
       f.Start.Accept (this); f.End.Accept (this); f.Body.Accept (this);
       return Void;
    }
-
+   
    public override NType Visit (NReadStmt r) {
-      throw new NotImplementedException ();
+      Visit ((IEnumerable<Node>)r.Vars[0]);

```

Comment:
r.Vars[0] will never be of type (IEnumerable<Node>). Did you check if this works?

Reply:
It's not working.  I just used the intellisense hints.